beancount/core/amount.py

```diff
@@ -161,10 +175,10 @@ def mul(amount, number):
     Returns:
       An Amount, with the same currency, but with 'number' times units.
     """
-    assert isinstance(amount.number, Decimal), (
-        "Amount's number is not a Decimal instance: {}".format(amount.number))
-    assert isinstance(number, Decimal), (
-        "Number is not a Decimal instance: {}".format(number))
+    assert isinstance(amount.number, (int, Decimal)), (
+        "Amount's number is not an Int or Decimal instance: {}".format(amount.number))
```

Comment:
The type name is `int` not `Int` but anyhow, `amount.number` is supposed to be `Decimal`, not `int`.

Reply:
fixed.